File: CivitasJava/src/civitas/CasillaSorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author ivan
 */
public class CasillaSorpresa extends Casilla {
     private Sorpresa sorpresa ;
     private MazoSorpresas mazo ;
     
     public CasillaSorpresa (MazoSorpresas mazo, String nombre) {
        super(nombre) ;
        this.mazo = mazo;
        this.sorpresa = mazo.siguiente() ;
    }
     
     @Override
    public String toString(){
        String s = super.toString() ;
        s+= "\nLa sorpresa que saldrá es:\n" + sorpresa.toString() ;
        return s;
    }
     
     @Override
    void recibeJugador(int actual, ArrayList<Jugador> todos){
        if(super.jugadorCorrecto(actual,todos)){
            super.informe(actual, todos);
           sorpresa = mazo.siguiente() ;
           this.informe(actual, todos);
           sorpresa.aplicarAJugador(actual, todos);
        }
    }
}

File: CivitasRuby/lib/CasillaSorpresa.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative "Sorpresa"


module Civitas
  class CasillaSorpresa < Casilla
    
    def initialize (mazo,nombre)
      @mazo = mazo
      @sorpresa = mazo.siguiente
      super(nombre)
    end
    
    def to_s
      s = super
      s += "\n La sorpresa que saldra es : #{@sorpresa.to_s}"
      return s
    end
    
    def recibeJugador(actual,todos)
      if(self.jugadorCorrecto(actual,todos))
        @sorpresa = @mazo.siguiente() 
        super
        @sorpresa.aplicarAJugador(actual, todos)
      end
    end
  end
end